api/src/main/java/org/pmiops/workbench/db/dao/DataSetService.java

```diff
@@ -32,4 +34,9 @@ DataSet saveDataSet(
       KernelTypeEnum kernelTypeEnum,
       String dataSetName,
       Map<String, QueryJobConfiguration> queryJobConfigurationMap);
+
+  DataSet cloneDataSetToWorkspace(
+      DataSet fromDataSet, Workspace toWorkspace, Set<Long> cohortIds, Set<Long> conceptSetIds);
+
+  List<DataSet> getDataSets(Workspace workspace);
```

Comment:
aside: This feels a bit like feature envy. I.e. it feels like it should live on the `Workspace` model. I understand that's not feasible in any of the workspace classes we have though.

Reply:
What Eric said :D 

I actually would have preferred it on the workspace model, so objects can fetch all of their information, but this seemed reasonable to me, and I did not have a strong opinion, so went with Eric's thoughts on it.